src/protocols/bdx/TransferFacilitator.cpp

```diff
@@ -71,28 +86,34 @@ void TransferFacilitator::PollTimerHandler(chip::System::Layer * systemLayer, vo
     static_cast<TransferFacilitator *>(appState)->PollForOutput();
 }
 
+void TransferFacilitator::ImmediatePollTimerHandler(chip::System::Layer * systemLayer, void * appState)
+{
+    // Restart polling with regular timeout
+    systemLayer->CancelTimer(ImmediatePollTimerHandler, appState);
+    VerifyOrReturn(appState != nullptr);
+    static_cast<TransferFacilitator *>(appState)->SchedulePoll();
+
+    // Do the job
+    TransferFacilitator::PollTimerHandler(systemLayer, appState);
+}
+
 void TransferFacilitator::PollForOutput()
 {
     TransferSession::OutputEvent outEvent;
     mTransfer.PollOutput(outEvent, System::SystemClock().GetMonotonicTimestamp());
     HandleTransferSessionOutput(outEvent);
+}
```

Comment:
So I feel like I am missing something here.  PollTimerHandler used to call PollForOutput, which would re-schedule the poll timer as needed.  What re-schedules that timer in the new setup?

Reply:
You are right. I've just over engineered a little bit. Fixed now.